core/music/Queue.js

```diff
@@ -0,0 +1,230 @@
+const m3u8stream = require('m3u8stream');
+const ytdl = require('ytdl-core');
+const EventEmitter = require('node:events');
+const { AudioPlayerStatus, createAudioPlayer, joinVoiceChannel, NoSubscriberBehavior, VoiceConnectionStatus } = require('@discordjs/voice');
+const { shuffleArray } = require('../../utility/Function');
+
+class Queue extends EventEmitter {
+    constructor(client, guildId, player) {
+        super();
+
+        this.AudioPlayer = createAudioPlayer({
+            behaviors: {
+                noSubscriber: NoSubscriberBehavior.Pause,
+            },
+        });
+
+        this.client = client;
+        this.guildId = guildId;
+        this.player = player;
+
+        this.AudioPlayer.on('stateChange', (oldState, newState) => {
+            if (newState.status === AudioPlayerStatus.Idle && oldState.status !== AudioPlayerStatus.Idle) {
+                if (this.loop === 0) {
+                    if (this.history.length > 5) {
+                        this.history.pop();
+                    }
+                    this.history.unshift(this.actualTrack);
+                }
+                setTimeout(() => {
+                    this.play();
+                }, 1000);
+            }
+        });
+
+        this.AudioPlayer.on('error', (error) => {
+            console.log(error);
+        });
+
+
+        this.queue = [];
+        this.history = [];
+        this.actualResource = null;
+        this.actualTrack = null;
+        this.loop = 0;
+        this.volume = 0.25;
+        this.musicChannel = null;
+        this.playing = false;
+    }
+
+    connect(channel) {
+
+        if (!channel) {
+            this.queue = [];
+            return false;
+        }
+
+        if (this.connection?.state?.status === 'ready') return false;
```

Comment:
This returns false when trying to add a song with youtube, having a song already playing
 => emits an error "channel not found" and doesnt add song to queue